File: Examples/ObjectiveC/VASTExample/VASTExample/VideoAdDescriptionViewController.m
/*
 * Version for iOS © 2015–2019 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

#import <YandexMobileAds/YandexMobileVASTAds.h>
#import "VideoAdDescriptionViewController.h"
#import "TrackingTableViewController.h"

@interface VideoAdDescriptionViewController ()

@property (nonatomic, strong, readonly) YMAVASTAd *ad;

@end

@implementation VideoAdDescriptionViewController

- (instancetype)initWithAd:(YMAVASTAd *)ad
{
    self = [super init];
    if (self != nil) {
        _ad = ad;
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.title = @"Video ads";
    
    UITextView *descriptionTextView = [[UITextView alloc] initWithFrame:CGRectZero];
    descriptionTextView.translatesAutoresizingMaskIntoConstraints = NO;
    descriptionTextView.editable = NO;
    descriptionTextView.text = self.ad.description;
    [self.view addSubview:descriptionTextView];
    NSDictionary *views = NSDictionaryOfVariableBindings(descriptionTextView);
    NSArray *horizontal = [NSLayoutConstraint constraintsWithVisualFormat:@"H:|[descriptionTextView]|"
                                                                  options:0
                                                                  metrics:nil
                                                                    views:views];
    NSArray *vertical = [NSLayoutConstraint constraintsWithVisualFormat:@"V:|[descriptionTextView]|"
                                                                options:0
                                                                metrics:nil
                                                                  views:views];
    [self.view addConstraints:[horizontal arrayByAddingObjectsFromArray:vertical]];
    
    self.navigationItem.rightBarButtonItem =
        [[UIBarButtonItem alloc] initWithTitle:@"Tracking"
                                         style:UIBarButtonItemStylePlain
                                        target:self
                                        action:@selector(displayTracking)];
}

- (void)displayTracking
{
    TrackingTableViewController *controller = [[TrackingTableViewController alloc] init];
    controller.ad = self.ad;
    [self.navigationController pushViewController:controller animated:YES];
}

@end

File: Examples/Swift/VASTExample/VASTExample/VideoAdDescriptionViewController.swift
/*
 * Version for iOS © 2015–2021 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

import YandexMobileAds

class VideoAdDescriptionViewController: UIViewController {
    private let ad: YMAVASTAd
    
    init(ad: YMAVASTAd) {
        self.ad = ad
        super.init(nibName: nil, bundle: nil)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.title = "Video ads"

        let descriptionTextView = createDescriptionTextView()
        view.addSubview(descriptionTextView)
        configureConstraint(descriptionTextView: descriptionTextView)
        configureNavigationItem()
    }

    private func configureNavigationItem() {
        navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Tracking",
                                                            style: .plain,
                                                            target: self,
                                                            action: #selector(displayTracking))
    }

    private func createDescriptionTextView() -> UITextView {
        let descriptionTextView = UITextView(frame: .zero)
        descriptionTextView.translatesAutoresizingMaskIntoConstraints = false
        descriptionTextView.isEditable = false
        descriptionTextView.text = ad.description
        return descriptionTextView
    }

    private func configureConstraint(descriptionTextView : UIView) {
        let views = ["descriptionTextView": descriptionTextView]
        let horizontal = NSLayoutConstraint.constraints(withVisualFormat: "H:|[descriptionTextView]|",
                                                        options: [],
                                                        metrics: nil,
                                                        views: views)
        let vertical = NSLayoutConstraint.constraints(withVisualFormat: "V:|[descriptionTextView]|",
                                                      options: [],
                                                      metrics: nil,
                                                      views: views)
        view.addConstraints(horizontal + vertical)
    }
    
    @objc private func displayTracking() {
        let controller = TrackingTableViewController(ad: ad)
        navigationController?.pushViewController(controller, animated: true)
    }
}
